### PR TITLE
Reorganised CFamily configuration code

### PR DESCRIPTION
**Pull request by duncanp-sonar**
* moved `Capture` class to separate file
* moved `Capture`-related methods from `FileConfig.cs` to `Capture.cs`
* moved `Capture`-related tests from to new test class

* no functional changes

### CODE REVIEW

File: src/Integration.Vsix/CFamily/CFamilyHelper.cs [hunk 1]
```diff
@@ -341,25 +349,5 @@ private static void Add(IList<string> cmd, string prefix, string value)
                 cmd.Add(value);
             }
         }
-
-        internal class Capture
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CFamilyHelper.cs:345]: `FileHelper` has two internal classes, `FileConfig` and `Capture`. I'm not sure what the point of `Capture` is. However, I've separate out the Capture-related code and tests into separate files to make it easier to see what is going on.

File: src/Integration.Vsix/CFamily/CFamilyHelper.cs [hunk 2]
```diff
@@ -309,6 +309,14 @@ internal static bool IsFileInSolution(ProjectItem projectItem)
             return false;
         }
 
+        private static Request ToRequest(FileConfig fileConfig, string path)
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CFamilyHelper.cs:312]: Moved from `FileConfig` and made into a static method. I also changed the scope from `public` to `private`.

File: src/Integration.Vsix.UnitTests/CFamily/FileConfigTests.cs [hunk 3]
```diff
@@ -116,85 +113,7 @@ public void GetPotentiallyUnsuppertedPropertyValue_CriticalException_IsNotSuppre
         }
 
         [TestMethod]
-        public void PlatformName()
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/FileConfigTests.cs:119]: These tests are all capture-related.

File: src/Integration.Vsix/CFamily/CFamilyHelper.cs (hunk 1 above)
Comment (rita-g-sonarsource) [reply src/Integration.Vsix/CFamily/CFamilyHelper.cs:345]: I believe Capture is now the class that contains the post-processing logic mentioned in https://github.com/SonarSource/sonarlint-visualstudio/issues/1524

**Pull request merged**